mountpoint-s3-client/src/endpoint_config.rs

```diff
@@ -148,10 +152,57 @@ impl EndpointConfig {
         let resolved_endpoint = endpoint_rule_engine
             .resolve(endpoint_request_context)
             .map_err(EndpointError::UnresolvedEndpoint)?;
-        let endpoint_uri = resolved_endpoint.get_url();
-        Uri::new_from_str(&allocator, endpoint_uri)
+
+        Ok(ResolvedEndpointInfo(resolved_endpoint))
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ResolvedEndpointInfo(ResolvedEndpoint);
+
+impl ResolvedEndpointInfo {
+    pub fn get_uri_from_endpoint(&self) -> Result<Uri, EndpointError> {
+        let allocator = Allocator::default();
+        let endpoint_url = self.0.get_url();
+        Uri::new_from_str(&allocator, endpoint_url)
             .map_err(|e| EndpointError::InvalidUri(InvalidUriError::CouldNotParse(e)))
     }
+
+    pub fn get_auth_scheme_from_endpoint(&self) -> Result<AuthScheme, EndpointError> {
+        let endpoint_properties = self.0.get_properties();
```

Comment:
Might be useful to add a comment inside this method with an example of an output from the resolver, so people reading the code know what the shape of it is.